### DISCUSSION TITLE
What if I have a large number of members in the struct

### DISCUSSION THREAD

**Discussion by dongmingsun** [nlohmann/json#3271] [Q&A] [answered]
According to [Simplify your life with macros](https://github.com/nlohmann/json#simplify-your-life-with-macros), to make my struct eligible for serialization and deserialization, I'd like to use `NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE` to save myself some time.

If I understand it right, the macro supports most **64** members? But the number of members in my struct is larger than that limit, and it is really a pain to hand-write the `to_json` and `from_json` functions.

Is there any convenient way to get this job done? Or hand-write the two functions is a must?

**Comment by nlohmann** [comment, marked as answer]
The code for the macros has been generated with this script: https://github.com/nlohmann/json/blob/develop/third_party/macro_builder/main.cpp. It should be possible to extend it to a larger number of members. What number would you need?

**Reply by dongmingsun** [reply]
137 for now, and the number may grow even further.
Thanks for the script provided; I'll update the json.hpp used in my project manually.
